src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -580,6 +601,40 @@ void Gather::exec1DCase() {
     }
 }
 
+void Gather::execCompressedEmbedding() {
+    DEBUG_LOG(getName(), " execCompressedEmbedding");
+    auto srcMemPtr = getParentEdgeAt(GATHER_DATA)->getMemoryPtr();
+    auto idxMemPtr = getParentEdgeAt(GATHER_INDICES)->getMemoryPtr();
+    const auto* psrc = reinterpret_cast<const uint8_t*>(srcMemPtr->getData());
+    const auto* pidx = reinterpret_cast<const int32_t*>(idxMemPtr->getData());
+
+    const auto* zp = reinterpret_cast<const float_t*>(decompressionSubtractPtr->getData());
+    const auto* scale = reinterpret_cast<const float_t*>(decompressionMultiplyPtr->getData());
+
+    auto* pdst = reinterpret_cast<float_t*>(getChildEdgeAt(0)->getMemoryPtr()->getData());
+
+    const auto& idxDims = idxMemPtr->getStaticDims();
+    const auto batch = idxDims[0];
+    const auto seqLen = idxDims[1];
+
+    auto axisDim = srcMemPtr->getStaticDims()[0];
+    auto feaDim = srcMemPtr->getStaticDims()[1];
+
+    parallel_for2d(batch, seqLen, [&](size_t b, size_t s) {
+        auto dstIdx = b * seqLen + s;
+        auto ii = pidx[dstIdx];
+        if (ii < 0) {
+            if (reverseIndexing)
+                ii += axisDim;
+            else
+                ii = axisDim;
+        }
+        for (size_t f = 0; f < feaDim; f++) {
+            pdst[dstIdx * feaDim + f] = (static_cast<float>(psrc[ii * feaDim + f]) - zp[ii]) * scale[ii];
+        }
```

Comment:
maybe the compiler you used is good enough to optimize this loop, but let's not take it for granted since we support many different platform and different compiler,  suggest to extract invariant out of the loop:
```C++
auto* src = psrc + ii*feaDim;
auto* dst = pdst + dstOdx*feaDim;
auto& deq_zp = zp[ii];
auto& deq_scale = scale[ii];
for (size_t k = 0; k < feaDim; k++) {
    dst[k] = (static_cast<float>(src[k]) - deq_zp) * deq_scale;
}
```

Reply:
Done